src/vos/evtree.c

```diff
@@ -2279,6 +2296,13 @@ evt_insert(daos_handle_t toh, const struct evt_entry_in *entry,
 
 	evt_ent_array_init(ent_array, 1);
 
+	/* For evt_remove_all, we only insert removals for things that
+	 * are visible, so we shouldn't need to check for overwrite at
+	 * all.
+	 */
```

Comment:
Don't we support inserting overlapping removal records?

Reply:
We first find any records that are visible and then insert a removal to cover them.   There should be no overlap with other removal records because anything under those would not be visible.

Reply:
Got it, thanks.